suite-common/wallet-core/src/transactions/transactionsThunks.ts

```diff
@@ -226,60 +230,43 @@ export const addFakePendingCardanoTxThunk = createThunk(
+    accountKey: AccountKey;
+    page: number;
+    perPage: number;
+    noLoading?: boolean;
+    forceRefetch?: boolean;
+};
+
+export const fetchTransactionsPageThunk = createThunk(
+    `${TRANSACTIONS_MODULE_PREFIX}/fetchTransactionsPageThunk`,
     async (
-        {
-            accountKey,
-            page,
-            perPage,
-            noLoading = false,
-            recursive = false,
-        }: {
-            accountKey: AccountKey;
-            page: number;
-            perPage: number;
-            noLoading?: boolean;
-            recursive?: boolean;
-        },
-        { dispatch, getState, signal },
+        { accountKey, page, perPage, forceRefetch }: FetchTransactionsPageThunkParams,
+        { dispatch, getState },
     ) => {
+        // console.log('fetchTransactionsPageThunk', accountKey, page, perPage, forceRefetch);
         const account = selectAccountByKey(getState(), accountKey);
-        if (!account) return;
-        if (!isTrezorConnectBackendType(account.backendType)) return; // skip unsupported backend type
-        const transactions = selectTransactions(getState());
-        const reducerTxs = getAccountTransactions(account.key, transactions);
+        if (!account) {
+            throw new Error(`Account not found: ${accountKey}`);
+        }
+        if (!isTrezorConnectBackendType(account.backendType)) {
+            throw new Error(`Unsupported backend type: ${account.backendType}`);
+        }
 
+        const transactions = selectAccountTransactionsWithNulls(getState(), account.key); // get all transactions including "null" values because of pagination
         const startIndex = (page - 1) * perPage;
         const stopIndex = startIndex + perPage;
-        const txsForPage = reducerTxs.slice(startIndex, stopIndex).filter(tx => !!tx.txid); // filter out "empty" values
-
-        // we already got txs for the page in reducer
-        if (
-            (page > 1 && txsForPage.length === perPage) ||
-            txsForPage.length === account.history.total
-        ) {
-            if (recursive && !signal.aborted && account.history.total) {
-                const promise = dispatch(
-                    fetchTransactionsThunk({
-                        accountKey,
-                        page: page + 1,
-                        perPage,
-                        noLoading,
-                        recursive: true,
-                    }),
-                );
-                signal.addEventListener('abort', () => {
-                    promise.abort();
-                });
-                await promise;
-            }
-
-            return;
-        }
+        const txsForPage = transactions.slice(startIndex, stopIndex).filter(tx => !!tx.txid); // filter out "empty" values
+        // always fetch first page because there might be new transactions
+        const isFirstPage = page === 1;
+        const isPageAlreadyFetched = txsForPage.length === perPage;
```

Comment:
Good catch but not sure how to handle it better (if needed). It will be issue only on first page (which is refeteched always anyway) and on the last page.

Reply:
Hmm, for the first page only if it is a last page as well. You are right, this is probably unnecessary to change, as it will happen only once per account max.